app/src/main/java/com/scrumbums/donationboi/controllers/ItemSearchActivity.java

```diff
@@ -106,4 +107,32 @@ private void setListeners(Button searchButton) {
         });
     }
 
+    private static final double FUZZYTHRESH = 0.7;
+
+    /**
+     * Do a fuzzy match search of two Strings.
+     * @param actual The value gotten.
+     * @param expected The expected value.
+     * @return If the given string matches more than a given percent of the expected String.
+     */
+    private static boolean fuzzyContains(CharSequence actual, CharSequence expected) {
+        double matching = 0;
+        int expIndex = 0;
+        for (int i = 0; i < actual.length() && expIndex < expected.length(); i++) {
+            char act = actual.charAt(i);
+            char exp = expected.charAt(expIndex);
+            if (Character.isAlphabetic(act)) {
+                act = Character.toLowerCase(act);
+            }
+            if (Character.isAlphabetic(exp)) {
+                exp = Character.toLowerCase(exp);
+            }
+            if (act == exp) {
+                matching++;
+                expIndex++;
+            }
+        }
+        return matching / expected.length() > FUZZYTHRESH;
```

Comment:
Is it possible expected.length() will ever be 0?  Just think checking for divide by 0 would be a good idea

Reply:
double division by 0 is POSITIVE_INFINITY which is defined to be greater than any other double and thus would automatically add the item. also expected is the search bar text and testing it, it looks like it will short-circuit on the contains("").